src/main/java/gregtech/api/recipes/ingredients/IntCircuitIngredient.java

```diff
@@ -27,9 +56,54 @@ protected IntCircuitIngredient copy() {
         return copy;
     }
 
-    public IntCircuitIngredient(int matchingConfigurations) {
-        super(getIntegratedCircuit(matchingConfigurations));
-        this.matchingConfigurations = matchingConfigurations;
+    @Override
+    public GTRecipeInput copyWithAmount(int amount) {
+        return copy(); // Amount of IntCircuitIngredient is always 1
+    }
+
+    @Override
+    public GTRecipeInput setNBTMatchingCondition(NBTMatcher nbtMatcher, NBTCondition nbtCondition) {
+        return this; // IntCircuitIngredient ignores nbt conditions
+    }
+
+    @Override
+    public int getAmount() {
+        return 1;
+    }
+
+    @Override
+    public ItemStack[] getInputStacks() {
+        if (this.inputStacks == null) {
+            this.inputStacks = new ItemStack[]{getIntegratedCircuit(this.matchingConfigurations)};
+        }
+        return this.inputStacks;
+    }
+
+    @Override
+    public boolean acceptsStack(@Nullable ItemStack itemStack) {
+        return itemStack != null && MetaItems.INTEGRATED_CIRCUIT.isItemEqual(itemStack) &&
+                matchingConfigurations == getCircuitConfiguration(itemStack);
+    }
+
+    @Override
+    public int hashCode() {
+        return Objects.hash(matchingConfigurations, isConsumable);
```

Comment:
This `hashCode` implementation should be calculated and stored separately, as it will be hashed very often.

Reply:
Actually for `IntCircuitIngredient` the hashCode isn't even necessary anymore, because it doesn't use the cache hashtable.

Regardless, I'll try to implement hash caching for `GTRecipeInput`s.